src/components/FilterPanel/index.tsx

```diff
@@ -126,8 +165,112 @@ class FilterPanel extends React.Component<Props, States> {
     changeFilter(selectedEntity, selectedFilter, index);
   };
 
+  onFilterAttributeChange = (val, index) => {
+    const { filters, selectedEntity, changeFilter } = this.props;
+    const selectedFilter: any = filters[index];
+    selectedFilter.attribute = val;
+    changeFilter(selectedEntity, selectedFilter, index);
+  };
+
+  onValueChange = val => {
+    const { setValue } = this.props;
+    setValue(val);
+  };
+
+  generateValues = items => {
+    let newItems = [];
+    items.map((item, index) => {
+      if (item !== null) {
+        const items = item.replace(/(^|_)./g, s =>
+          s
+            .split('_')
+            .map(s => s.charAt(0).toUpperCase() + s.substring(1))
+            .join(' ')
+        );
+        newItems.push(items);
+      } else if (item === null) {
+        newItems.push('Null');
+      }
+    });
+    return newItems;
+  };
+
+  generateFilter = filter => {
+    const { values, attributes, value, selectedEntity } = this.props;
+    const menuValues =
+      (selectedEntity === 'operations' && filter.name === 'kind') ||
+      (selectedEntity === 'operations' && filter.name === 'status') ||
+      (selectedEntity === 'operations' && filter.name === 'spendable') ||
+      (selectedEntity === 'operations' && filter.name === 'delegatable') ||
+      (selectedEntity === 'accounts' && filter.name === 'spendable') ||
+      (selectedEntity === 'accounts' && filter.name === 'delegate_setable')
+        ? this.generateValues(values)
```

Comment:
Is there an automated way of pulling the names of filters with less than 15 cardinality than hardcoding? Maybe we can add a function to ConseilJS to do this for us?

Also, it seems like it is missing `protocol` &` chain_id ` from `Blocks` entity

Reply:
protocol & chain_id's values don't require any sort of manipulation so that's why they're left out here. I could probably write a function that checks for an underscore and then runs if it finds one

Reply:
entity metadata includes cardinality per field, so it can be generated programmatically

Reply:
^ most recent updates have excluded this hardcoded checking, thank you @anonymoussprocket @umurb